src/mapeo-project.js

```diff
@@ -667,15 +668,32 @@ export class MapeoProject extends TypedEmitter {
     // 2. Assign LEFT role for device
     await this.#roles.assignRole(this.#deviceId, LEFT_ROLE_ID)
 
-    // 3. Clear data from cores
-    // TODO: only clear synced data
+    // 3. Clear data
+    await this[kClearDataIfLeft]()
+  }
+
+  /**
+   * Clear data if we've left the project. No-op if you're still in the project.
+   * @returns {Promise<void>}
+   */
+  async [kClearDataIfLeft]() {
+    const role = await this.$getOwnRole()
+    if (role.roleId !== LEFT_ROLE_ID) {
+      return
+    }
+
     const namespacesWithoutAuth =
       /** @satisfies {Exclude<import('./core-manager/index.js').Namespace, 'auth'>[]} */ ([
         'config',
         'data',
         'blob',
         'blobIndex',
       ])
+    const dataStoresToUnlink = Object.values(this.#dataStores).filter(
+      (dataStore) => dataStore.namespace !== 'auth'
+    )
+
+    await Promise.all(dataStoresToUnlink.map((ds) => ds.close()))
```

Comment:
We need to close earlier in the process now, because of the following race condition:

1. We open a project.
1. We close the writer core (see line 636 below).
1. We start indexing the writer core (via `MultiCoreIndexer`). _This fails because the writer core has been closed._

(This was theoretically a problem before. Someone could load a project and immediately leave it.)